GoogleSignIn/Sources/GIDAppCheck/Implementations/GIDAppCheck.h

```diff
@@ -19,18 +19,45 @@
 #if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
 
 #import <Foundation/Foundation.h>
-#import "GoogleSignIn/Sources/GIDAppCheck/API/GIDAppCheckProvider.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-@class FIRAppCheckToken;
+@protocol GACAppCheckProvider;
+@protocol GACAppCheckTokenProtocol;
+
 extern NSString *const kGIDAppCheckPreparedKey;
 
 NS_CLASS_AVAILABLE_IOS(14)
-@interface GIDAppCheck : NSObject <GIDAppCheckProvider>
```

Comment:
Can we keep this `GIDAppCheckProvider` protocol and use its fake in `GIDSignInTest`? That'd make `GIDSignInTest` just test `GIDSignIn`'s implementation (and not `GIDAppCheck`'s).

Reply:
I don't think it's necessary for `GIDAppCheckProvider` to exist any longer since it is more or less redundant on `AppCheckCore`'s own protocol: `GACAppCheckProvider`. `GIDAppCheck` takes an instance of some type conforming to `GACAppCheckProvider`, which is used by `AppCheckCore`'s `GACAppCheck` type to get the limited use tokens.

With regard to `GIDSignInTest`, I think it's preferable to test with real types and not fakes/stubs/mocks since we typically use the latter when the former are not practical or possible (or are costly for some reason). Since a real instance of `GIDAppCheck` is possible to construct and use in tests (with the help of providing a fake conforming to `GACAppCheckProvider` to simulate errors or successful token retrieval), I think it makes sense to use `GIDAppCheck`.

Let me know if I've misunderstood your recommendation.